### PR TITLE
refactor: cleanup ccv types

### PR DESCRIPTION
**Pull request by bermuell**
## Description

Closes: #1213

Refactor shared events, codecs and errors assign to consumer and provider dedicated types where possible
Please note that it was agreed to no add the split out of the `expected_keepers.go` in this PR.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if state-machine breaking change (i.e., requires coordinated upgrade)
* [x] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [x] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release for both the consumer and provider

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,7 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/tidwall/gjson v1.17.0
 	golang.org/x/crypto v0.11.0 // indirect
-	golang.org/x/exp v0.0.0-20230711153332-06a737ee72cb
```
Comment (MSalopek) [inline go.mod:26]: How did this happen?

Did you `go mod tidy`?

File: x/ccv/provider/keeper/proposal_test.go [hunk 2]
```diff
@@ -231,6 +231,7 @@ func TestPendingConsumerAdditionPropDeletion(t *testing.T) {
 	defer ctrl.Finish()
 
 	for _, tc := range testCases {
+		tc := tc
```
Comment (MSalopek) [inline x/ccv/provider/keeper/proposal_test.go:234]: Not sure this is actually an issue?

Can you confirm?

**Review by MSalopek — APPROVED** [review]
LGTM!

Please check if `tc := tc` is actually needed in some of the tests?
I think that becomes an issue only when you're storing pointers in arrays/slices. In the testcases you changed the slice contains objects, and not pointers to objects.

I could be wrong, but I played with this a bit:
* https://go.dev/play/p/TZ2TIntvXc7

**Review by p-offtermatt — APPROVED** [review]
LGTM

File: x/ccv/provider/keeper/proposal_test.go (hunk 2 above)
Comment (bermuell) [reply x/ccv/provider/keeper/proposal_test.go:234]: not an issue considering how the data is in SetPendingConsumerAdditionProp used but this is a strong assumption and could change. It's safer this way IMO.

File: go.mod (hunk 1 above)
Comment (bermuell) [reply go.mod:26]: yes I ran go mod tidy which added it

**Pull request merged**